azure/resource-containers/pf-resource-group/functions/public/get-pfresourcegroup.ps1

```diff
@@ -0,0 +1,15 @@
+Function Get-PfResourceGroup{
+    Param (
+        [Parameter(Mandatory = $false)] 
+        [string] $Name,
+        [Parameter(Mandatory = $false)] 
+        [string] $SubscriptionName
+    )
+
+    $contextResourceGroupName = Get-PfAzureContextResourceGroupName -Name $Name
+    $contextSubscriptionName = Get-PfAzureContextSubscriptionName -Name $SubscriptionName
+
+    $access = (New-Object -TypeName "PfResourceGroupAccess")
+    return $access.GetCloudState($contextResourceGroupName,$contextSubscriptionName)
+}
```

Comment:
should this return a new 'definition' object, with cloudstate + placeholder for build state?

Reply:
I'm starting to wonder the benefit of definition now.  Only using the function "New-PfKeyvault" will add to a deployment context.  So not sure bringing the build state back makes sense.  Starting to now think we just user build and cloud states separately as I am not sure when we would need them together in an object as they will already be defined in the script.  Thoughts?